source/server/listener_manager_impl.cc

```diff
@@ -281,111 +292,215 @@ bool ListenerImpl::isWildcardServerName(const std::string& name) {
   return absl::StartsWith(name, "*.");
 }
 
-void ListenerImpl::addFilterChain(const std::vector<std::string>& server_names,
+void ListenerImpl::addFilterChain(uint16_t destination_port,
+                                  const std::vector<std::string>& destination_ips,
+                                  const std::vector<std::string>& server_names,
                                   const std::string& transport_protocol,
                                   const std::vector<std::string>& application_protocols,
                                   Network::TransportSocketFactoryPtr&& transport_socket_factory,
                                   std::vector<Network::FilterFactoryCb> filters_factory) {
   const auto filter_chain = std::make_shared<FilterChainImpl>(std::move(transport_socket_factory),
                                                               std::move(filters_factory));
-  // Save mappings.
+  addFilterChainForDestinationPorts(filter_chains_, destination_port, destination_ips, server_names,
+                                    transport_protocol, application_protocols, filter_chain);
+}
+
+void ListenerImpl::addFilterChainForDestinationPorts(
+    DestinationPortsMap& destination_ports_map, uint16_t destination_port,
```

Comment:
filter_chains_ is converted to destination_ports_map ?

Reply:
Not converted, `filter_chains_` is of the `DestinationPortsMap` type.

Reply:
is it possible to rename it for clarity? :).